cloudify/workflows/workflow_context.py

```diff
@@ -1622,17 +1622,27 @@ def __init__(self, workflow_ctx, modification):
         self._raw_modification = modification
         self.workflow_ctx = workflow_ctx
         node_instances = modification.node_instances
-        added_raw_nodes = dict(
-            (instance.node_id, workflow_ctx.get_node(instance.node_id)._node)
-            for instance in node_instances.added_and_related).values()
+        added_raw_nodes = []
+        seen_ids = set()
+        for instance in node_instances.added_and_related:
+            if instance.node_id not in seen_ids:
+                added_raw_nodes.append(
+                    workflow_ctx.get_node(instance.node_id)._node)
+                seen_ids.add(instance.node_id)
+
```

Comment:
this (+below) could've been simpler
```python
added_raw_nodes = [
    workflow_ctx.get_node(instance.node_id)._node
    for instance in node_instances.added_and_related
]
```
however the old code also did deduplication by node id, so we're going to do it too, but more explicitly (you can now actually see which items have precedence)